### PR TITLE
change Data texture defaults

### PR DESCRIPTION
**Pull request by gero3**
adapt DataTexture based on #7062

### CODE REVIEW

File: examples/js/SimulationRenderer.js [hunk 1]
```diff
@@ -206,7 +206,6 @@ function SimulationRenderer( WIDTH, renderer ) {
 		texture.minFilter = THREE.NearestFilter;
```
Comment (mrdoob) [inline examples/js/SimulationRenderer.js]: Should these be removed too?

File: examples/js/SimulationRenderer.js [hunk 2]
```diff
@@ -232,7 +231,6 @@ function SimulationRenderer( WIDTH, renderer ) {
...
```
Comment (mrdoob) [inline examples/js/SimulationRenderer.js]: And these...?

File: examples/js/postprocessing/GlitchPass.js [hunk 3]
```diff
@@ -113,7 +113,6 @@ THREE.GlitchPass.prototype = {
 		texture.minFilter = THREE.NearestFilter;
```
Comment (mrdoob) [inline examples/js/postprocessing/GlitchPass.js]: And... these? 😊

File: examples/js/SimulationRenderer.js (hunk 1 above)
Comment (gero3) [inline examples/js/SimulationRenderer.js]: indeed Fixed

File: examples/js/SimulationRenderer.js (hunk 2 above)
Comment (gero3) [inline examples/js/SimulationRenderer.js]: indeed fixed

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
All these methods with many arguments will need to be cleaned up at some point...

**Comment by WestLangley** [comment]
Do you mean this API?

    var tex = new THREE.DataTexture( { data: data, width: width, height: height } );

Or maybe this?

    var tex = new THREE.DataTexture( data, width, height, optionalParams );

**Comment by mrdoob** [comment]
The later, but lets wait a bit... 😁

**Comment by mrdoob** [comment]
Should we also set `WebGLRenderTarget.generateMipmaps` default to `false`?
https://github.com/mrdoob/three.js/blob/dev/src/renderers/WebGLRenderTarget.js#L32

**Comment by WestLangley** [comment]
> Should we also set WebGLRenderTarget.generateMipmaps default to false?

That sounds right to me. The examples would have to be carefully checked and corrected, however -- a good idea in any event.

**Comment by EliasHasle** [comment]
I ran into this upgrade today. I had created a very simple texture in mspaint, and decided it would be more fun to generate it in JS, so I created a DataTexture with default settings, and it rendered horribly. I found out from the docs that some settings were not Texture defaults, but frankly the docs aren't very clear on what default settings, among those mentioned, are different from the Texture defaults.

> Further parameters correspond to the properties inherited from Texture, where both magFilter and minFilter default to THREE.NearestFilter.
In my ears this sounds like:
> Among other defaults, THREE.NearestFilter is the default minFilter and magFilter for Texture, and this is inherited by DataTexture.

This turns out not to be the case. So after finally digging up the default filters from the Texture docs, I tried selecting them for the DataTexture. Ooops! Have to be really careful to have big Ms in *both* "Mip" and "Map"... OK. Then a lot of error messages. Aha! There is a binary flag for mipmaps too, that apparently differs from the default. So I set *generateMipMaps=true*. *Very* careful to have a big large M in "Maps", yes... Same error messages again, but of course none to tell me that the flag is called "generateMip**m**aps" (finding out that solved my problem).

Having the same defaults whenever possible makes the system predictable, which is good for newbies, and I thought it would be good for experts too. Changing the defaults is kind of a job for people who (think they) know what they are doing. But then again, I am kind of voluntarily joining that group by using DataTexture.

**Comment by EliasHasle** [comment]
Ouch, that was an angry old comment. But 2016 Elias had some points. I had a similar experience today, although some of the problems, namely the inconsistent capital 'M', have been solved.

I note that when a DataTexture is configured with one of the Mipmap filters, it still needs generateMipmaps=true (or manual mipmaps) to work. Would it be possible to modify the behavior so that mipmaps are generated when needed on the first render?

**Pull request merged**